src/main/java/au/org/ala/biocache/service/AuthService.java

```diff
@@ -184,26 +193,111 @@ public void run() {
         } else {
             thread.run();
         }
-
     }
 
-    public List<String> getUserRoles(String userId) {
-        List<String> roles = new ArrayList<>();
-        if(StringUtils.isNotBlank(userDetailsUrl)) {
+    @Deprecated
+    public Set<String> getUserRoles(String userId) {
```

Comment:
Similar to `getUserDetails`, This is marked as `@Deprecated` but still seems to be used for downloads ?
This results in 401 since the request to the user details endpoint does not include any authentication parameter. This should be updated to use JWT or API key authentication.

Reply:
this method is only used when configured `auth.legacy.emailonly.downloads.enabled` which should be deprecated.